fish-rust/src/curses.rs

```diff
@@ -271,3 +317,24 @@ impl FlagCap {
         FlagCap(Code::new(code))
     }
 }
+
+/// Covers over tparm().
+pub fn tparm1(cap: &CStr, param1: i32) -> Option<CString> {
+    // Take the lock because tparm races with del_curterm, etc.
+    let _term: std::sync::MutexGuard<Option<Arc<Term>>> = TERM.lock().unwrap();
+    let cap_ptr = cap.as_ptr() as *mut libc::c_char;
+    // Safety: we're trusting tparm here.
+    unsafe {
+        // Check for non-null and non-empty string.
+        assert!(!cap_ptr.is_null() && cap_ptr.read() != 0);
```

Comment:
I think `cap_ptr` can never be null (should be enforced by CStr)
so how about `assert!(!cap.to_bytes().is_empty());`

Reply:
done